src/components/sound.js

```diff
@@ -24,44 +23,33 @@ module.exports.Component = registerComponent('sound', {
 
   update: function (oldData) {
     var data = this.data;
-    var diffData = diff(oldData || {}, data);
     var el = this.el;
     var sound = this.sound;
-    var src = data.src;
-    var srcChanged = 'src' in diffData;
+    var srcChanged;
+
+    oldData = oldData || {};
```

Comment:
let's do an empty object for empty data and add the flag later if there's need. if a conponent wants to do something at inititiization the method init should be used